gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -6,25 +6,48 @@ module T::Private::Methods
   @signatures_by_method = {}
   @sig_wrappers = {}
   @sigs_that_raised = {}
+  # the info about whether a method is final is not stored in a DeclBuilder nor a Signature, but instead right here.
+  # this is because final checks are special:
+  # - they are done possibly before any sig block has run.
+  # - they are done even if the method being defined doesn't have a sig.
```

Comment:
is this true?

Reply:
Ah, in this PR the only way you can mark a method as final is with its sig, so no. but, in #1178, I add final modules, and then in that case, you can have a final module with a method, and the final checks are run for that method even if there's no sig. So yes.

I rebased such that #1178 now is a descendant of this PR - I can rebase #1178 onto master if this PR gets merged into master first.